src/utils.jl

```diff
@@ -1,3 +1,13 @@
+"""
+    pick_chunksize()
+
+Return a reasonable chunk size for batched differentiation, given as `Val{chunk}()`.
+
+!!! warning
+    This function is experimental, and not part of the public API.
+"""
+pick_chunksize() = Val(16)
```

Comment:
Can this take a mode, function type, and argument types?

Reply:
and also possibly a val for the total number of iterations

Reply:
what do you mean by iterations?

Reply:
Like we’re chunking a total of N iterations into chunks or size C

Reply:
are argument types sufficient? in general we may need the actual argument values, for instance to ensure that the chunk size is smaller than the input length

Reply:
Yeah that’s why I also recommended passing in the total number of iterations too

Reply:
perhaps `dimension` is a better word?

Reply:
maybe but that feels ambiguous in a different way

Reply:
Why does this need to return a `Val`?

Reply:
Because Enzyme takes this parameter as a `Val` for `jacobian` and (forward) `gradient`, so I thought this would save the eventual users one conversion. In addition, because it is constant, it is also type-inferrable (for now).

Reply:
>  In addition, because it is constant, it is also type-inferrable (for now).

If it ever becomes non-constant this will be a source of latency. Julia will constant-evaluate `Val(chunksize())` just fine, but the Val conversion should be left to the user.

Reply:
I added the arguments and a `Val` called `totalsize`, is that how you imagined it?